blackrock_data_fetcher.py

```diff
@@ -66,38 +62,26 @@ def create_local_directories(today):
     return d
 
 
-def fetch_file(remote_dir, local_dir):
-    if DEBUG:
-        print("Fetching %s to %s" % (remote_dir, local_dir))
-
-    cmd = '%s -oPort=%s %s@%s:"%s" %s ' % (
-        SCP, SFTP_PORT, SFTP_USER, SFTP_HOST, remote_dir, local_dir)
-    if DEBUG:
-        print("cmd: %s" % (cmd))
-
-    child = pexpect.spawn(cmd)
-    child.expect('password:')
-    child.sendline(SFTP_PASSWD)
-    # make sure to extend the timeout for long download times. 5
-    # min should do it.
-    child.expect(pexpect.EOF, timeout=600)
-
-
-def fetch_files(remote_dir, local_dir):
+def fetch_files(local_dir):
     if DEBUG:
-        print("Fetching %s to %s" % (remote_dir, local_dir))
-
-    cmd = '%s -oPort=%s %s@%s:"%s/*" %s ' % (
-        SCP, SFTP_PORT, SFTP_USER, SFTP_HOST, remote_dir, local_dir)
-    if DEBUG:
-        print("cmd: %s" % (cmd))
-
-    child = pexpect.spawn(cmd)
-    child.expect('password:')
-    child.sendline(SFTP_PASSWD)
-    # make sure to extend the timeout for long download times. 5
-    # min should do it.
-    child.expect(pexpect.EOF, timeout=600)
+        print("Fetching from Google Drive to %s" % (local_dir))
+    virtual_forest = get_api_response(METADATA_URI, VIRTUAL_FOREST_ID)
+    items = virtual_forest.json()['files']
+    for item in items:
+        # If the item is a folder, add the folder contents to the item list
+        if item['mimeType'] == 'application/vnd.google-apps.folder':
+            subfolder = get_api_response(METADATA_URI, item['id'])
+            for subitem in subfolder.json()['files']:
+                items.append(subitem)
```

Comment:
Noting the approach here of adding files to the `items` list. In general, modifying a structure which is currently being iterated can result in unexpected side effects. (The Python doc notes that it is "[tricky to get right](https://docs.python.org/3/tutorial/controlflow.html#for-statements)".) 

Suggestion would be to have the items iteration create a secondary "to be downloaded" list of files. This approach would also separate functionality, e.g. iterating the drive and downloading the files.